micro-core/src/main/java/com/aol/micro/server/utility/HashMapBuilder.java

```diff
@@ -25,6 +28,15 @@ public Builder(K key, V value) {
 			build = new HashMap<K, V>();
 			build.put(key, value);
 		}
+		public Builder(Map<K,V> map) {
+			build = new HashMap<K, V>();
```

Comment:
From official documentation: "An instance of HashMap has two parameters that affect its performance: initial capacity and load factor. [...] When the number of entries in the hash table exceeds the product of the load factor and the current capacity, the hash table is rehashed [...] If many mappings are to be stored in a HashMap instance, creating it with a sufficiently large capacity will allow the mappings to be stored more efficiently than letting it perform automatic rehashing as needed to grow the table."

If we are building here a copy of the passed map, I would suggest to initialize it with the proper size.